### PR TITLE
Adjust Connection signature to match `sqlite3.connect`

### PR DESCRIPTION
**Pull request by thedrow**

### CODE REVIEW

File: src/pyrqlite/connections.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ class Connection(object):
     )
 
     def __init__(self, scheme='http', host='localhost', port=4001,
-                 user=None, password=None, connect_timeout=None,
+                 user=None, password=None, timeout=None,
                  detect_types=0, max_redirects=UNLIMITED_REDIRECTS):
```
Comment (zmedico) [inline src/pyrqlite/connections.py:45]: If we remove the `connect_timeout` parameter then also sqlalchemy-rqlite will need to be updated here:

https://github.com/rqlite/sqlalchemy-rqlite/blob/8aa51797ac72523361f3bf531b48f3ea6232d0b3/src/sqlalchemy_rqlite/pyrqlite.py#L76

**Comment by thedrow** [comment]
There is also `isolation_level` which is missing. If it's not relevant, maybe we should accept kwargs and ignore them.

**Comment by zmedico** [comment]
> There is also `isolation_level` which is missing. If it's not relevant, maybe we should accept kwargs and ignore them.

If it's recognized, then yes it's fine to ignore. If it's unrecognized then it's better to report it with warnings.warn, in order to help correct mistakes.

File: src/pyrqlite/connections.py (hunk 1 above)
Comment (zmedico) [reply src/pyrqlite/connections.py:45]: Added backward compat and deprecated it in a8e25175d03290fcba6237de0dbf6c6a1892a436.

**Comment by zmedico** [comment]
> There is also `isolation_level` which is missing. If it's not relevant, maybe we should accept kwargs and ignore them.

Also in python 3.12 they've added an `autocommit` parameter and connection attribute which they suggest to set to False in order to enable [PEP 249](https://peps.python.org/pep-0249/)-compliant transaction control:

https://docs.python.org/3.12/library/sqlite3.html#transaction-control

Prior to python 3.12 there's just `isolation_level` for transaction control:

https://docs.python.org/3.11/library/sqlite3.html#transaction-control

**Pull request closed without merging**